### ISSUE TITLE
Add rule `no-useless-children-snippet`

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/eslint-plugin-svelte#909]
### Motivation

Any content inside a component tag becomes the `children` snippet. Thus, If you have a component with a children snippet without parameters, that's unnecessary.

### Description

I'd like to add a rule that bans the `children` snippet if it has no parameters

### Examples

```svelte
<script>
</script>

<!-- ✓ GOOD -->
<Foo>
  {#snippet bar()}
    Hello
  {/snippet}
</Foo>

<Foo>
  {#snippet children(val)}
    Hello {val}
  {/snippet}
</Foo>

<Foo>
  Hello
</Foo>

<!-- ✗ BAD -->
<Foo>
  {#snippet children()}
    Hello
  {/snippet}
</Foo>
```

**Comment by ota-meshi** [comment]
Thank you for proposing the rule. SGTM!

Maybe the rule should be a bit clearer. For example, how about the name `no-useless-children-snippet`?
I think the rule name might confuse users if we later create a rule to report components like this:

```svelte
{#snippet foo(value)}
  Hello {value}
{/snippet}

<!-- ✗ BAD -->
<Foo>
  {#snippet bar(value)}
    {@render foo(value)}
  {/snippet}
</Foo>

<!-- ✓ GOOD -->
<Foo bar={foo}>
</Foo>
```

**Comment by marekdedic** [comment]
Makes sense, renaming! Thanks :)
